distributed/comm/core.py

```diff
@@ -278,16 +278,18 @@ def _raise(error):
     if timeout and timeout / 20 < backoff:
         backoff = timeout / 20
 
-    retry_timeout_backoff = random.randrange(140, 160) / 100
+    retry_timeout_backoff = max(5, timeout / 4) * random.uniform(0.9, 1.1)
 
     # This starts a thread
     while True:
         try:
             while deadline - time() > 0:
-
-                async def _():
-                    comm = await connector.connect(
-                        loc, deserialize=deserialize, **connection_args
+                with suppress(TimeoutError):
```

Comment:
Why that? If the server socket is saturated/unresponsive, this will timeout, hit the `break` and further spam with connection requests without any backoff.

Reply:
Yeah, that's part of the logic bug I mentioned. The other logic has to do with failures/timeouts in the handshake (which would lead to your handshake errors). I'm not sure what the cause of your other errors is.

Reply:
Not sure on the cause of the 2nd error ether. My intuituon is that os may have run out of ports. Happy to give it an other try later. Thx for looking into this:))